bycycle/tests/group/test_utils.py

```diff
@@ -0,0 +1,21 @@
+"""Test group utility functions."""
+
+import time
+import pytest
+import numpy as np
+
+from bycycle.group.utils import progress_bar
+
+###################################################################################################
+###################################################################################################
+
+@pytest.mark.parametrize("progress", [None, 'tqdm',
+                                      pytest.param('invalid', marks=pytest.mark.xfail)])
+def test_progress_bar(progress):
+
+    n_iterations = 10
+    iterable = [0] * n_iterations
+
+    pbar = progress_bar(iterable, progress, n_iterations)
+
+    assert type(pbar) == list
```

Comment:
This test fails for me locally. I currently have tqdm version 4.47.0. 

I get:
`AssertionError: assert <class 'tqdm.std.tqdm'> == list where <class 'tqdm.std.tqdm'>`

I _think_ this is actually right - when `tqdm` is used, then `pbar` is not type list, and so we need to update this test. 

I think it doesn't fail on travis but we haven't added `tqdm` to any dependencies, so it doesn't actually get run using a progress bar?

Maybe though there is some kind of version difference somewhere. Does this test pass for you locally?

Reply:
This needs to be updated. It passes for me locally only when I don't have tqdm installed in my environment. I'll fix this.